providers/base/bin/gateway_ping_test.py

```diff
@@ -474,52 +400,72 @@ def main(argv) -> int:
+    host = get_host_to_ping(interface=args.interfaces[0], target=args.host)
+
+    print(_("Checking connectivity to {0}").format(host))
 
     if host:
-        ping_summary = ping(
-            host, args.interface, args.count, args.deadline, args.verbose
-        )
+        ping_summary = ping(host, args.interfaces[0])
     else:
         ping_summary = {
             "received": 0,
             "cause": "Unable to find any host to ping",
         }
 
     if ping_summary["received"] == 0:
-        print(_("No Internet connection"))
+        print(_("FAIL: All packet loss."))
         if ping_summary.get("cause"):
             print("Possible cause: {}".format(ping_summary["cause"]))
         return 1
     elif ping_summary["transmitted"] != ping_summary["received"]:
-        print(
-            _("Connection established, but lost {0}% of packets").format(
-                ping_summary["pct_loss"]
-            )
-        )
-        if ping_summary["pct_loss"] > args.threshold:
-            print(
-                _(
-                    "FAIL: {0}% packet loss is higher than {1}% threshold"
-                ).format(ping_summary["pct_loss"], args.threshold)
-            )
-            return 1
-        else:
-            print(
-                _("PASS: {0}% packet loss is within {1}% threshold").format(
-                    ping_summary["pct_loss"], args.threshold
-                )
-            )
-            return 0
+        print(_("FAIL: {0}% packet loss.").format(ping_summary["pct_loss"]))
+        return 1
     else:
-        print(_("Connection to test host fully established"))
+        print(_("PASS: 0% packet loss").format(host))
         return 0
 
 
+def get_default_gateways() -> Dict[str, str]:
+    """
+    Use `route` program to find default gateways for all interfaces.
+
+    returns a dictionary in a form of {interface_name: gateway}
+    """
+    try:
+        routes = subprocess.check_output(
+            ["route", "-n"], universal_newlines=True
+        )
+    except subprocess.CalledProcessError as exc:
+        logging.debug("Failed to run `route -n `", exc)
+        return {}
+    regex = r"^0\.0\.0\.0\s+(?P<gw>[\w.]+)\s.*\s(?P<interface>[\w.]+)$"
+    matches = re.finditer(regex, routes, re.MULTILINE)
+
+    return {m.group("interface"): m.group("gw") for m in matches}
+
+
+def is_cable_interface(interface: str) -> bool:
+    """
+    Check if the interface is a cable interface.
+    This is a simple heuristic that checks if the interface is named
+    "enX" or "ethX" where X is a number.
+
+    :param interface: the interface name to check
+    :return: True if the interface is a cable interface, False otherwise
+
+    Looking at the `man 7 systemd.net-naming-scheme` we can see that
+    even the `eth` matching may be an overkill.
+    """
+    if not isinstance(interface, str) or not interface:
+        return False
+    return interface.startswith("en") or interface.startswith("eth")
```

Comment:
It is a bit weird, given the `Route` structure, that these are free functions

Reply:
I don't know how I feel about using `class`es as namespaces. ATM, for me the less digging, the less indent, the better.